deep_qa/training/run_model.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import shutil
+import os
+
+import random
+import pyhocon
+import numpy
+
+# pylint: disable=wrong-import-position
+sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+from deep_qa.common.params import Params, replace_none, ConfigurationError
+from deep_qa.common.tee_logger import TeeLogger
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+def prepare_environment(params: Params):
+    seed = params.pop("random_seed", 13370)
+    numpy_seed = params.pop("numpy_seed", 1337)
+    if "keras" in sys.modules:
+        logger.warning("You have already imported Keras in your code. If you are using DeepQA"
+                       "functionality to set random seeds, they will have no effect, as code"
+                       "prior to calling this function will be non-deterministic. We will not"
+                       "the random seed here.")
+        seed = None
+        numpy_seed = None
+    if seed is not None:
+        random.seed(seed)
+    if numpy_seed is not None:
+        numpy.random.seed(numpy_seed)
+
+
+def run_model(param_path: str):
+
+    """
+    This function is the normal entry point to DeepQA. Use this to run a DeepQA model in
+    your project. Note that if you care about exactly reproducible experiments,
+    you should avoid importing Keras before you import and use this function, as
+    Keras relies on random seeds which can be set in this function via a
+    JSON specification file.
+
+    Parameters
+    ----------
+    param_path: str, required. A json file specifying a DeepQaModel.
+    """
+    param_dict = pyhocon.ConfigFactory.parse_file(param_path)
+    params = Params(replace_none(param_dict))
+    prepare_environment(params)
+
+    # These have to be imported _after_ we set the random seed,
+    # because keras uses the numpy random seed.
+    from deep_qa.common.checks import log_keras_version_info
+    from deep_qa.models import concrete_models
+    from keras import backend as K
+
+    log_keras_version_info()
+    log_dir = params.get("model_serialization_prefix", None)  # pylint: disable=no-member
+    if log_dir is not None:
+        sys.stdout = TeeLogger(log_dir + "_stdout.log", sys.stdout)
+        sys.stderr = TeeLogger(log_dir + "_stderr.log", sys.stderr)
+        handler = logging.FileHandler(log_dir + "_python_logging.log")
+        handler.setLevel(logging.INFO)
+        handler.setFormatter(logging.Formatter('%(asctime)s - %(levelname)s - %(name)s - %(message)s'))
+        logging.getLogger().addHandler(handler)
+        shutil.copyfile(param_path, log_dir + "_model_params.json")
+    model_type = params.pop_choice('model_class', concrete_models.keys())
+    model_class = concrete_models[model_type]
+    model = model_class(params)
+
+    if model.can_train():
+        logger.info("Training model")
+        model.train()
+        K.clear_session()
+    else:
+        raise ConfigurationError("The supplied model does not have enough training inputs.")
```

Comment:
`model.train()` already runs tests if you give test data, right?  Maybe we should split that out, but that's something for another PR, if we want to do it.